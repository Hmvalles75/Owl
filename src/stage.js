import React from 'react';
import StageParticipants from './StageParticipants';

export default function(props){
  return (
      <div className='stageList'>
          {props.participants.map(item => (
          <StageParticipant key={item.id} name={item.name} avatar={item.avatar} 
          inSession={item.inSession} onStage={item.onStage} />))}
      </div>
  );
}
