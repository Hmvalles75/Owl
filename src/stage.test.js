import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import stage from './stage';

it ('renders without crashing', () => {
  const div = document.createElement('div')
  ReactDOM.render(<stage participants={[
    {
      id: 1,
      name: 'Koren Templeton',
      avatar:
          'https://robohash.org/itaquevoluptatumrerum.jpg?size=200x200&set=set1',
      inSession: true,
      onStage: true
  },
  {
      id: 3,
      name: 'Axe Kubicka',
      avatar: 'https://robohash.org/utlaborumfugit.jpg?size=200x200&set=set1',
      inSession: false,
      onStage: false
  },
  ]} />, div)
  ReactDOM.unmountComponentAtNode(div);
});

it('Renders the UI as expected', () => {
  const tree = renderer
      .create(<stage participants={[
          {
              id: 1,
              name: 'Koren Templeton',
              avatar:
                  'https://robohash.org/itaquevoluptatumrerum.jpg?size=200x200&set=set1',
              inSession: true,
              onStage: true
          },
          {
              id: 3,
              name: 'Axe Kubicka',
              avatar: 'https://robohash.org/utlaborumfugit.jpg?size=200x200&set=set1',
              inSession: false,
              onStage: false
          },
      ]} />)
      .toJSON();
  expect(tree).toMatchSnapshot();
});



