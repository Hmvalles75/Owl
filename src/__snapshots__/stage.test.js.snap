// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders the UI as expected 1`] = `
<stage
  participants={
    Array [
      Object {
        "avatar": "https://robohash.org/itaquevoluptatumrerum.jpg?size=200x200&set=set1",
        "id": 1,
        "inSession": true,
        "name": "Koren Templeton",
        "onStage": true,
      },
      Object {
        "avatar": "https://robohash.org/utlaborumfugit.jpg?size=200x200&set=set1",
        "id": 3,
        "inSession": false,
        "name": "Axe Kubicka",
        "onStage": false,
      },
    ]
  }
/>
`;
